### ISSUE TITLE
Subsume INF225 Glossary

### ISSUE THREAD

**Issue by grammarware** [slebok/slebok#5]
See https://bitbucket.org/sle-uib/inf225/wiki/glossary/Glossary

**Comment by grammarware** [comment]
@anyahelene

**Comment by anyahelene** [comment]
So, what's best of just importing the MD files, and also leaving the source document (tex) so that we may regenerate things a few times until we decide on the exact format?

**Comment by anyahelene** [comment]
Have dumped the glossary into the repo now. Can still do some auto-reformatting until we start editing things.

**Comment by steimann** [comment]
Thanks Anya, great contribution!

**Comment by grammarware** [comment]
Proposed changes (also have a look at the commit above with an example):
* not capitalise the file name at all (to have something like `extended Backus-Naur form.md`)
* format Wikipedia links with `sameAs`
* one tag per line (we'll figure out later what we do with tags)
* `Tag Index.md` is not really needed, we can generate things like that for the website and keep the repo free of them
* fix comma-separated terms like `Structure, Structure type.md` names

Otherwise, seems not only great but almost perfect!

**Comment by anyahelene** [comment]
### FYI:  Information that is in the glossary source:
* definition text
* alternative names for the same term (this *includes* stuff like things that are closely related, so it's not directly importable to "Also known as")
* tags (a very restricted set of them)
* link to wikipedia
* abbreviation
* other links
* whether it is considered important or outside the scope for INF225 (not included in the SLEBOK dump)

I've put the available info in separate sections, apart from the abbreviation(s) which appear in parenthesis after the tile. If the information is not in the source document, it doesn't appear at all.

**Comment by anyahelene** [comment]
> Proposed changes (also have a look at the commit above with an example):
> 
>     not capitalise the file name at all (to have something like extended Backus-Naur form.md)

Should have very clear guidelines if that's the case. Is it EBNF = "Extended Backus-Naur Form" or ENBF = "extended Backus-Naur form"? 

* Is the rule to only capitalize given names?

* **Problem:** There are loads of internal links that may break.

>     format Wikipedia links with sameAs

' This is easily done. Will they be clickable? (Autolinking of URLs?)

>     one tag per line (we'll figure out later what we do with tags)

* Possible, but needs horrible TeX code. Perhaps easer to fix when we have a tool to manipulate the entriess?

>     Tag Index.md is not really needed, we can generate things like that for the website and keep the repo free of them

* Same goes for Glossary.md and other maybe other stuff.

>     fix comma-separated terms like Structure, Structure type.md names

* Yes, this is a case of talking about multiple things in one entry. Needs splitting or a solution.

> Otherwise, seems not only great but almost perfect!

* Perfection is a process, not a state.

**Comment by anyahelene** [comment]
On 23. aug. 2017 09:43, Vadim Zaytsev wrote:
> re: autolinking
> 
> I think actually the stupid solution would work great: add bare URLs, 
> they will be clickable from the repo right away, and it is always 
> possible to detect the website from the webfront-generating beautifier 
> which can add icons and other stuff.

Well, we'll still need to rewrite the existing entries. And of course 
validate (or even automigrate) the links if we do more renamings later.

Is there a fancy reason why GitHub doesn't format the links now? Is it 
by design, or a mistake in the syntax?

-anya

**Comment by anyahelene** [comment]
> I can reformat existing entries, no problem. Just waiting for you to say your fancy import is complete ;)
> 
> The [...](...) syntax only works for actual links, and GitHub cannot possibly guess that you are referring to a file with an .md extension

Right. The files are from Bitbucket Wiki, which is why it worked before.

**Comment by anyahelene** [comment]
Relative links work in GitHub (since 2013), but doesn't support spaces. I.e., spaces must be escaped with %20, or we just replace the them consistently with underscores (or just drop spaces in entry file names).

Question: What is preferable of:

1. Escaping spaces in links with %20, and keeping spaces in the file names
2. Replacing spaces with underscore
3. Removing spaces (would probably conflict with downcasing all the terms)

**Comment by grammarware** [comment]
I was surprised that spaces actually worked, and let this moment of surprise prevent me from objecting. Of course, it would make many things much simpler if we just had underscores and in general tried to avoid weird stuff in names.

**Comment by anyahelene** [comment]
So, any instance of a significant underscore? Otherwise I'd be happy to go with that solution, even though it's a bit 1970. Would make things easier if we built a tool chain with bash/sed/etc.

**Comment by gorelhedin** [comment]
I agree that it is a good idea to avoid blanks in names. Underscores may look ugly, but would be more practical. I can't think of any significant underscores for the glossary at least.
